applications/GeoMechanicsApplication/custom_processes/apply_excavation_process.cpp

```diff
@@ -0,0 +1,61 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Lorenzo Gracia,
+//                   Aron Noordam,
+//                   Vahid Galavi
+//                   Marjan Fathian
+//
+#include "apply_excavation_process.hpp"
+#include "includes/model_part.h"
+#include "includes/element.h"
+#include "utilities/variable_utils.h"
+#include "includes/kratos_flags.h"
+#include "includes/kratos_parameters.h"
+
+namespace Kratos
+{
+
+ApplyExcavationProcess::ApplyExcavationProcess(ModelPart&        rModelPart,
+                                               const Parameters& rProcessSettings)
+           : Process(Flags()),
+             mrModelPart{rModelPart}
+{
+    Parameters params = Parameters("{}");
+    params.AddValue("model_part_name", rProcessSettings["model_part_name"]);
+    params.AddValue("variable_name", rProcessSettings["variable_name"]);
+
+    if (rProcessSettings.Has("deactivate_soil_part")) {
+        params.AddValue("deactivate_soil_part", rProcessSettings["deactivate_soil_part"]);
+        KRATOS_TRY
+        mDeactivateSoilPart = rProcessSettings["deactivate_soil_part"].GetBool();
+        KRATOS_CATCH("")
+    }
```

Comment:
I feel that this part of the _original Python code_ was very misleading. The reason is that when "deactivate_soil_part" is not found in the settings, it did not create an internal process (i.e. `self.process` would not be defined). However, the other methods of the class assumed it was there. In other words, this class becomes useless when "deactivate_soil_part" is not found and it should result in an error immediately.

Furthermore, I noticed that `params` is built up, but it is never actually used. So we can remove that.

The only thing that is left is to initialize the data member `mDeactivateSoilPart`, which can be done using the class's initializer list. As a result, the body of the constructor becomes empty. Nice and clean :-)